app/src/main/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModel.kt

```diff
@@ -19,15 +23,28 @@ import kotlinx.coroutines.launch
 @HiltViewModel
 class PartiesViewModel @Inject constructor(
 	private val partiesRepository: PartiesRepository,
-) : ViewModel() {
+) : MVIViewModel<PartiesEvent>(),
+	SideEffectEmitter<PartiesSideEffect> by SideEffectEmitterImplementation() {
 
 	val state: MutableState<ViewModelState<List<Party>>> = mutableStateOf(Loading)
 
 	init {
 		loadParties()
 	}
 
-	fun loadParties() {
+	override fun handleEvent(event: PartiesEvent) {
+		when (event) {
+			is PartiesEvent.Load -> loadParties()
+		}
+	}
+
+	fun onPartyClick(partyId: Int) {
+		viewModelScope.launch {
+			setEffect(PartiesSideEffect.NavigateToPartyDetails(partyId))
+		}
+	}
+
+	private fun loadParties() {
 		viewModelScope.launch {
 			state.value = Loading
```

Comment:
Shouldn't we return the state from the repository? So the `Loading`, `Error` and `Success` with data. If I'm not wrong the data layer should return some kind of state to the ViewModel, and then it could be mapped in ViewModel.

Reply:
hmm, I am not sure about that :D I see this just like in MVVM, that we start with a loading state and then setting error/loaded with data, like i am doing it here :) There is also a way to change the state model itself a little bit, and just, for example, have a separate class for every screen, which has data in it, for example isLoading parameter etc. but i am not sure if it is a good solution :D